CSYS-210: Migrate Tags

CavalcanteLucas opened the pull request:
**Link to task:**
https://labcodes.atlassian.net/browse/CSYS-210

**Staging app:** https://migrate-tags--confetti-storybook.netlify.app/

**How to test:**
Open the staging app, then check if if the story of `SimpleTag`, `DropdownTag`, `RemovableTag` and `TogglableTag` work as expected.

**Description of your solution:**
This PR adds the stories of Tag components.

falconedani commented on stories/Tag/DropdownTag.stories.js:
```diff
@@ -0,0 +1,25 @@
+      control: { type: "select", options: TAG_COLORS },
+    },
+    skin: {
+      control: { type: "select", options: TAG_SKINS },
```
use radio instead, please.

CavalcanteLucas replied:
ok, fixed

falconedani commented on stories/Tag/RemovableTag.stories.js:
```diff
@@ -0,0 +1,43 @@
+      control: { type: "select", options: TAG_COLORS },
+    },
+    skin: {
+      control: { type: "select", options: TAG_SKINS },
```
use radio instead, please.

CavalcanteLucas replied:
ok, fixed

falconedani commented on stories/Tag/SimpleTag.stories.js:
```diff
@@ -0,0 +1,28 @@
+      control: { type: "select", options: TAG_COLORS },
+    },
+    skin: {
+      control: { type: "select", options: TAG_SKINS },
```
use radio instead, please.

CavalcanteLucas replied:
ok, fixed

falconedani reviewed (changes requested):
All seems to be working fine, although some details before merging:
- Use radio instead of select for skin prop. Please always use radio control when less than 4 options are available
- Use a boolean for avatar

**QA issues, only adjust if easy enough:**
- Icon display an empty value, replace with either `none` or `empty`
- Is it possible that the restore button on RemovableTag only display after the tag is removed?
- When I choose both icon and avatar I get `RemovableTag can't be initialized with both thumb and icon props.` but this is irreversible, even if I reset the props it doesn't go away. Is there a way to avoid that?

CavalcanteLucas commented:
> All seems to be working fine, although some details before merging:
> 
> * Use radio instead of select for skin prop. Please always use radio control when less than 4 options are available

OK!

> * Use a boolean for avatar

I don't think that would be a good solution in this case, because avatar is **not** a boolean field. I believe that it might lead to confusion. 
I'd suggest two solutions:
1 - increase the number of available thumb sources examples, so we make a proper use of select;
2 - give the user a hint. Something like: "Use this as an example of `thumbSrc`: <link.to.our.thumb>"

> **QA issues, only adjust if easy enough:**
> 
> * Icon display an empty value, replace with either `none` or `empty`

Removed the `empty` option. The user can either refresh the page or reload the args to remove the icon.

> * Is it possible that the restore button on RemovableTag only display after the tag is removed?

Yes, fixed.

> * When I choose both icon and avatar I get `RemovableTag can't be initialized with both thumb and icon props.` but this is irreversible, even if I reset the props it doesn't go away. Is there a way to avoid that?

I don't think we could "reverse" the error state one it's reached. 
I *think* we could build a stateful logic to conditionally disable controls so that can never happen.
But both the error and the documentaion are clear enough IMO.

falconedani reviewed (changes requested):
> I don't think that [use a boolean for avatar] would be a good solution in this case, because avatar is not a boolean field. I believe that it might lead to confusion.

I do agree with what you've said, but the approach used is not good for the experience and it's very confusing. I've thought it through and I believe a better approach is to use radios instead. The first option as `None`, the second `Labcodes logo`.

- Icon display an empty value, replace with either `none` or `empty`

> Removed the empty option. The user can either refresh the page or reload the args to remove the icon.

This approach goes against the purpose of using Storybook's controls. Please follow the instruction.

lucianoratamero commented:
hey @falconedani , I have some input on your comments.

> > I don't think that [use a boolean for avatar] would be a good solution in this case, because avatar is not a boolean field. I believe that it might lead to confusion.
> 
> I do agree with what you've said, but the approach used is not good for the experience and it's very confusing. I've thought it through and I believe a better approach is to use radios instead. The first option as `None`, the second `Labcodes logo`.

while the site for documentation can be more free with the approach on the controls (like we are on the playgrounds), on the storybook side, we need to be more closely tied to what the component is, code-wise. since `thumbSrc` is a string field on the component, presenting it as a boolean or select field will give the developers the wrong impression that the field is not of the string type. becase of that, and since the storybook is aimed at developers, I tend to agree with @CavalcanteLucas , even if it's not the best user experience.

> 
>     * Icon display an empty value, replace with either `none` or `empty`
> 
> 
> > Removed the empty option. The user can either refresh the page or reload the args to remove the icon.
> 
> This approach goes against the purpose of using Storybook's controls. Please follow the instruction.

this is the same issue as what we discussed above. on the react level, the `icon` prop needs to be one of the options available as an Icon type, and Icons don't have an empty option as a type. I know this is not the best experience for non-developers, but I really prefer to be faithful with how the components are implemented on the storybook side.

lucianoratamero reviewed (approved):
LGTM!
